### PR TITLE
[Mobile] List/quote: Unwrap inner block when pressing Backspace at start

### PR DESCRIPTION
**Pull request by geriux**
## What?
This PR is a follow-up of https://github.com/WordPress/gutenberg/pull/45075 to make these changes work on mobile.

## Why?
To have the same behavior in both web and mobile when unwrapping inner blocks when pressing `Backspace` at the start of the block.

## How?
By bringing the code changes to the `onMerge` functionality in the Block list. It also adds the `List Item` block in the mobile `transformationCategories`.

It also updates a test that has an outdated functionality with the current one.

## Testing Instructions

- Open the app
- Add a list block
- Add two items
- Select the first item and move the cursor to the start
- Press the `Backspace` key
- **Expect** the List item to be a Paragraph block outside of the List block

Repeat the same with the `Quote` block.

## Screenshots or screencast 

List block | Quote block
-|-
<kbd></kbd> | <kbd></kbd>

### CODE REVIEW

**Review by ellatrix — APPROVED** [review]
Thank you!

**Review by derekblank — APPROVED** [review]
LGTM! I was able to test successfully. 🚀

**Comment by geriux** [comment]
Hey @ellatrix 👋 should I merge this branch into yours or wait until you merge your branch into `trunk`? Let me know, thanks!

**Comment by ellatrix** [comment]
@geriux You can merge it into my branch because I have failing tests without this, so can't merge into trunk rn. You have failing unit tests here though?

**Comment by geriux** [comment]
> @geriux You can merge it into my branch because I have failing tests without this, so can't merge into trunk rn. You have failing unit tests here though?

@ellatrix It's weird because I only changed `*.native.js` files. I tried running `npm run test:unit -- --ci --maxWorkers=2 --cacheDirectory="$HOME/.jest-cache"` locally in the base branch `try/list-quote-merge-gradually` and it fails, not sure why they passed in https://github.com/WordPress/gutenberg/pull/45075 🤔

**Comment by ellatrix** [comment]
Let's merge it into my branch and see what happens.

**Pull request merged**
